types/transaction.go

```diff
@@ -150,7 +151,13 @@ func (t *Transaction) Cost() *big.Int {
 //   - or calculate a value with formula: min(gasFeeCap, gasTipCap * baseFee);
 func (t *Transaction) GetGasPrice(baseFee uint64) *big.Int {
 	if (t.GasPrice != nil && t.GasPrice.BitLen() > 0) || baseFee == 0 {
-		return new(big.Int).Set(t.GasPrice)
```

Comment:
Is this same as
```
if (t.GasPrice != nil && t.GasPrice.BitLen() > 0) {
    return new(big.Int).Set(t.GasPrice)
} else if baseFee == 0  {
    return new(big.Int)
}
```
? somehow it is easier for me if there is one less if inside first if block